ocs_ci/ocs/platform_nodes.py

```diff
@@ -1668,42 +1668,91 @@ def start_nodes(self, nodes):
             "Start nodes functionality is not implemented"
         )
 
-    def restart_nodes(self, nodes, wait=True):
-        raise NotImplementedError(
-            "Restart nodes functionality is not implemented"
+    def restart_nodes(self, nodes, timeout=360, wait=True):
+        """
+        Restart Azure vm instances
+
+        Args:
+            nodes (list): The OCS objects of the nodes / Azure Vm instance
+            wait (bool): True if need to wait till the restarted node reaches
+                READY state. False otherwise
+            timeout (int): time in seconds to wait for node to reach 'not ready' state,
+                and 'ready' state.
+
+        """
+        vms = [n.name for n in nodes]
+        assert vms, (
+            f"Failed to get the Azure Vm instances for "
+            f"nodes {[n.name for n in nodes]}"
```

Comment:
Thanks for pointing it out Martin,

- Dropping the below statements (missed to remove this earlier), as we are dealing with node object and not the azure vm objects in this method

```
        assert vms, (
            f"Failed to get the Azure Vm instances for "
            f"nodes {[n.name for n in nodes]}"
```

- Included the check on nodes variable and raised ValueError in case of empty list being passed

- I don't think any exception will occur when on accessing name attribute of node object, so check for this is avoided